subprojects/core/src/main/java/org/gradle/initialization/GradleLauncher.java

```diff
@@ -26,23 +26,31 @@
 public interface GradleLauncher extends Stoppable {
 
     GradleInternal getGradle();
-    SettingsInternal getSettings();
 
     /**
-     * <p>Executes the build for this {@code GradleLauncher} instance and returns the result.</p>
+     * Evaluates the settings for this build.
      *
-     * @return The result. Never returns null.
+     * @return The loaded settings instance.
      * @throws ReportedException On build failure. The failure will have been logged.
      */
-    BuildResult run() throws ReportedException;
+    SettingsInternal getLoadedSettings();
+
+    /**
+     * Configures the build.
+     * This is different from {@link #getBuildAnalysis()} in that the `buildStarted` and `buildFinished` events will not be fired.
```

Comment:
Is this really the case? It seems like buildStarted will still be called at least.